### DISCUSSION TITLE
Compression only material

### DISCUSSION THREAD

**Discussion by scote89** [robbievanleeuwen/section-properties#71] [Ideas]
Is there anything planned to permit compression only behaviors when calculating the section properties (ex. convergence function)?

**Comment by Czarified** [comment]
Perhaps I'm not understanding, but this would really only apply when evaluating _stress_ on a section. Moments of Inertia/Area and geometric properties should have no change (for elastic analysis), based on the material properties.

You're looking for support of a compression-only material in the stress analysis steps?

**Reply by scote89** [reply]
Yes, you are right but it would also affect the plastic properties as it would consider tension yielding

**Reply by connorferster** [reply]
I would like this too, for analysis of reinforced concrete sections. @scote89, is this your application also or are you thinking of something more general? I have a though to create a reinforced concrete module but, like the plastic properties calculations, the basic features for reinforced concrete would not require using the FEM (i.e. biaxial interaction diagrams).

**Reply by scote89** [reply]
Yeah, that's pretty much what I have in mind as well.

**Reply by robbievanleeuwen** [reply]
This is definitely possible, and I would argue that the finite element approach used in _sectionproperties_ makes this **relatively** simple compared to other approaches.

Happy to form a working committee on specific material modules post v2 implementation! I'm seeing these modules more as libraries that could be separate to _sectionproperties_, but open to architectural discussions as that's more just a vibe than anything based on solid thinking.

Robbie

**Comment by scote89** [comment]
Now that reinforced concrete sections are more present and available to the users in v2, I think its very important to warn the user in the documents that tension in concrete is being considered which is generally not permitted in structural codes. A compression-only module would still be still VERY useful for Reinforced Concrete sections!
